lib/ansible/modules/cloud/azure/azure_rm_image_facts.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/python
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_image_facts
+
+version_added: "2.7"
+
+short_description: Get facts about azure custom images.
+
+description:
+    - List azure custom images. The images can be listed where scope of listing can be based on subscription, resource group, name or tags.
+
+options:
+    resource_group:
+        description:
+            - Name of resource group.
+    name:
+        description:
+            - Name of the image to filter from existing images.
+    tags:
+        description:
+            - List of tags to be matched.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Madhura Naniwadekar(@Madhura-CSI)"
+'''
+
+
+EXAMPLES = '''
+- name: List images with name
+  azure_rm_image_facts:
+    name: test-image
+    resource_group: test-resource-group
+
+- name: List images by resource group
+  azure_rm_image_facts:
+    resource_group: test-resource-group
+    tags:
+      - testing
+      - foo:bar
+
+- name: List all available images under current subscription
+  azure_rm_image_facts:
+'''
+
+
+RETURN = '''
+images:
+    description: List of image dicts.
+    returned: always
+    type: list
+    example: [{
```

Comment:
instead of example now we need proper description of returned values, please check facts modules like azure_rm_webapp_facts or azure_rm_containerregistry_facts as a reference